### PR TITLE
Tidy up profile start/stop

### LINKED ISSUE #197: Move cuda profiler blocks behind ifdefs
We already have a profile arg to dust_cuda_options, could define an ifdef like `DUST_ENABLE_PROFILE`, then move things (including the Dust destructor!) behind an ifdef

### PR DESCRIPTION
**Pull request by richfitz**
Quite a small quality of life improvement, mostly I like this because it removes the destructor.

Also makes the nvcc options fully configurable - we could theoretically remove the fast math one with this in, but I think that'll be fairly popular

Fixes #197

### CODE REVIEW

File: R/cuda.R [hunk 1]
```diff
@@ -117,6 +117,13 @@ dust_cuda_configuration <- function(path_cuda_lib = NULL,
 ##'   docs](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html)
 ##'   for more details.
 ##'
+##' @param flags Optional extra arguments to pass to nvcc. These
+##'   options will not be passed to your normal C++ compiler, nor the
+##'   linker (for that use R's user Makevars system). This can be used
+##'   to do things like tune the maximum number of registers
+##'   (`--maxregcount=x`). If not `NULL`, this must be a character
```
Comment (johnlees) [inline R/cuda.R:124]:
```suggestion
##'   (`--maxrregcount x`). If not `NULL`, this must be a character
```

(I don't know why...)

File: inst/include/dust/dust.hpp [hunk 2]
```diff
@@ -73,8 +73,10 @@ class Dust {
     }
   }
 
-// This could be removed/commented out eventually
-#ifdef __NVCC__
+  // We only need a destructor when running with cuda profiling; don't
+  // include ond otherwise because we don't actually follow the rule
+  // of 3/5/0
+#ifdef DUST_ENABLE_CUDA_PROFILER
```
Comment (johnlees) [inline inst/include/dust/dust.hpp:79]: Is the `#endif` missing here?

**Review by johnlees — CHANGES_REQUESTED** [review]

File: inst/include/dust/dust.hpp (hunk 2 above)
Comment (richfitz) [reply inst/include/dust/dust.hpp:79]: on line 83, no?

Comment (johnlees) [reply inst/include/dust/dust.hpp:79]: Yes, I think was just obscured by the diff!

**Review by johnlees — APPROVED** [review]

**Pull request merged**